jenkins_pipelines/environments/common/pipeline-build-validation.groovy

```diff
@@ -204,6 +205,38 @@ def run(params) {
                     client_stage_result_fail = true
                 }
             }
+
+            /** Salt migration stages begin **/
+            if (params.enable_salt_migration_stages) {
+                try {
+                    stage('Add MUs Salt migration') {
+                        if (params.must_add_MU_repositories && params.enable_salt_migration_stages) {
+                            if (params.confirm_before_continue) {
+                                input 'Press any key to start adding Maintenance Update repositories'
+                            }
+                            echo 'Add custom channels and MU repositories'
+                            res_mu_repos = sh(script: "./terracumber-cli ${common_params} --logfile ${resultdirbuild}/testsuite.log --runstep cucumber --cucumber-cmd '${env.exports} cd /root/spacewalk/testsuite; rake cucumber:build_validation_add_maintenance_update_repositories_salt_migration_minion'", returnStatus: true)
+                            echo "Custom channels and MU repositories status code: ${res_mu_repos}"
+                            sh "exit ${res_mu_repos}"
+                        }
+                    }
+                    stage('Bootstrap Salt migration minion') {
+                        if (params.must_boot_node && params.enable_salt_migration_stages) {
+                            if (params.confirm_before_continue) {
+                                input 'Press any key to start bootstraping the Salt migration minion'
+                            }
+                            echo 'Register Salt migration minion'
+                            res_init_salt_migration = sh(script: "./terracumber-cli ${common_params} --logfile ${resultdirbuild}/testsuite.log --runstep cucumber --cucumber-cmd '${env.exports} cd /root/spacewalk/testsuite; rake cucumber:build_validation_init_salt_migration'", returnStatus: true)
```

Comment:
If we keep this init client as it is, it will be automatically generated by the rake configuration. This test will be run in parallel with the other init clients and we will even have smoke test s run against it.